2_adventures/001/jonkernpa/04_day/README.md

```diff
@@ -0,0 +1,11 @@
+# Adventure Day 4
+
+Just what is an adventure?
+
+How lame... spent an hour cleaning up my lousy mess of git rebase hell.
+
+     git remote add upstream /url/to/original/repo
+     git fetch upstream
+     git checkout master
+     git reset --hard upstream/master
+     git push origin master --force
```

Comment:
This happens to all of us and sometimes more than once.  merge conflicts and rebase go hand in hand.

But rebase is not something to stay clear of, it is something that is very powerful and useful, and to keep in your tool box for managing your own personal history before you push that history out to the 'pubic' or remote repository that others have access to.

But this adventure, I think, you will remember for a long time.